service/s3.cc

```diff
@@ -2761,11 +2723,7 @@ std::shared_ptr<S3Api> getS3ApiForUri(const std::string & uri)
         accessKey = getAwsAccessKey(accessKeyId);
     }
 
-    auto api = make_shared<S3Api>(accessKeyId, accessKey);
```

Comment:
It's simpler if we throw on registration of an inaccessible bucket than when we access it.  I'd keep this in here.